File: 2015/9/9.js
var num = 142453146368;
var alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split("");
var digits = [];

do {
	num--;
	var rest = num % alphabet.length;
	digits.unshift(alphabet[rest]);
	num -= rest;
	num /= alphabet.length;
} while (num > 0);

console.log(digits.join(""));

File: 2015/9/9.hs
alphabet :: [Char]
alphabet = ['A'..'Z']

solve :: Int -> String -> String
solve 0 acc = acc
solve num acc = let num1 = num-1
                    rest = num1 `mod` length alphabet 
                    num2 = (num1 - rest) `div` length alphabet
                in  solve num2 ((alphabet !! rest):acc)

main = do
    putStrLn $ solve 142453146368 []
